## Fetch the artifact created (sample.csv) from W&B and read it with pandas

In [1]:
 import wandb
 import pandas as pd

 run = wandb.init(project="nyc_airbnb", group="eda", save_code=True)
 local_path = wandb.use_artifact("sample.csv:latest").file()
 df = pd.read_csv(local_path, index_col='id')

wandb: Currently logged in as: f4ec123. Use `wandb login --relogin` to force relogin


In [2]:
df

name    host_id  \
id                                                                       
9138664                 Private Lg Room 15 min to Manhattan   47594947   
31444015  TIME SQUARE CHARMING ONE BED IN HELL'S KITCHEN...    8523790   
8741020   Voted #1 Location Quintessential 1BR W Village...   45854238   
34602077  Spacious 1 bedroom apartment 15min from Manhattan  261055465   
23203149   Big beautiful bedroom in huge Bushwick apartment     143460   
...                                                     ...        ...   
7094539                   Tranquil haven in bubbly Brooklyn    2052211   
4424261                     Large 1 BR with backyard on UWS    3447311   
4545882                 Amazing studio/Loft with a backyard   23569951   
26518547          U2 comfortable double bed sleeps 2 guests     295128   
33631782               Private Bedroom in Williamsburg Apt!    8569221   

             host_name neighbourhood_group    neighbourhood  latitude  \
id                                                                      
9138664           Iris              Queens        Sunnyside  40.74271   
31444015        Johlex           Manhattan   Hell's Kitchen  40.76682   
8741020           John           Manhattan     West Village  40.73631   
34602077         Regan              Queens          Astoria  40.76424   
23203149         Megan            Brooklyn         Bushwick  40.69839   
...                ...                 ...              ...       ...   
7094539        Adriana            Brooklyn  Windsor Terrace  40.65360   
4424261          Sarah           Manhattan  Upper West Side  40.80188   
4545882          Kaveh           Manhattan  Upper East Side  40.78110   
26518547  Carol Gloria               Bronx     Clason Point  40.81225   
33631782          Andi            Brooklyn     Williamsburg  40.71829   

          longitude        room_type  price  minimum_nights  \
id                                                            
9138664   -73.92493     Private room     74               2   
31444015  -73.98878  Entire home/apt    170               3   
8741020   -74.00611  Entire home/apt    245               3   
34602077  -73.92351  Entire home/apt    125               3   
23203149  -73.92044     Private room     65               2   
...             ...              ...    ...             ...   
7094539   -73.97546  Entire home/apt    143              14   
4424261   -73.96808  Entire home/apt    200               2   
4545882   -73.94567  Entire home/apt    220               3   
26518547  -73.85502     Private room     80               1   
33631782  -73.95819     Private room    109               3   

          number_of_reviews last_review  reviews_per_month  \
id                                                           
9138664                   6  2019-05-26               0.13   
31444015                  0         NaN                NaN   
8741020                  51  2018-09-19               1.12   
34602077                  1  2019-05-24               0.65   
23203149                  8  2019-06-23               0.52   
...                     ...         ...                ...   
7094539                   2  2016-08-27               0.04   
4424261                  22  2019-05-21               0.50   
4545882                  28  2019-05-23               0.50   
26518547                  4  2019-07-01               1.48   
33631782                  3  2019-04-28               1.07   

          calculated_host_listings_count  availability_365  
id                                                          
9138664                                1                 5  
31444015                               1               188  
8741020                                1                 0  
34602077                               1                13  
23203149                               2                 8  
...                                  ...               ...  
7094539                    

In [9]:
!pip install -U pandas-profiling[notebook]

Defaulting to user installation because normal site-packages is not writeable
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 324.4/324.4 kB 1.5 MB/s eta 0:00:000:00:010:00:01:01
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 345.9/345.9 kB 9.2 MB/s eta 0:00:000:00:01
  Using cached ipywidgets-8.0.6-py3-none-any.whl (138 kB)
  Using cached widgetsnbextension-4.0.7-py3-none-any.whl (2.1 MB)
  Using cached jupyterlab_widgets-3.0.7-py3-none-any.whl (198 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 33.7/33.7 MB 14.0 MB/s eta 0:00:00m eta 0:00:010:00:01
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.1/12.1 MB 16.5 MB/s eta 0:00:00m eta 0:00:010:00:01
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 11.8/11.8 MB 15.6 MB/s eta 0:00:00m eta 0:00:010:00:01
  Using cached pydantic-1.10.7-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 102.7/102.7 kB 10.7 MB/s eta 0:00:00
  Using cached numpy-1.23.5-cp310-cp310-manylinux_2_17_x86_64.m

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 233.8/233.8 kB 14.2 MB/s eta 0:00:00
  Created wheel for htmlmin: filename=htmlmin-0.1.12-py3-none-any.whl size=27098 sha256=73d73c582bd77c3ae8f4d6ea056a6a01cc4107263ac1a378a61253e68377eef4
  Stored in directory: /home/hieu/.cache/pip/wheels/dd/91/29/a79cecb328d01739e64017b6fb9a1ab9d8cb1853098ec5966d
Successfully built htmlmin
  Attempting uninstall: tqdm
    Found existing installation: tqdm 4.65.0
    Uninstalling tqdm-4.65.0:
      Successfully uninstalled tqdm-4.65.0
  Attempting uninstall: numpy
    Found existing installation: numpy 1.24.3
    Uninstalling numpy-1.24.3:
      Successfully uninstalled numpy-1.24.3
  Attempting uninstall: scipy
    Found existing installation: scipy 1.10.1
    Uninstalling scipy-1.10.1:
      Successfully uninstalled scipy-1.10.1
  Attempting uninstall: pandas
    Found existing installation: pandas 2.0.1
    Uninstalling pandas-2.0.1:
      Successfully uninstalled pandas-2.0.1
  Attempting uninstall: ma

## Create a profile with pandas_profiling

In [3]:
import pandas_profiling

profile = pandas_profiling.ProfileReport(df)
profile.to_widgets()

/tmp/ipykernel_8749/2516835558.py:1: DeprecationWarning: `import pandas_profiling` is going to be deprecated by April 1st. Please use `import ydata_profiling` instead.
  import pandas_profiling


Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render widgets:   0%|          | 0/1 [00:00<?, ?it/s]

## Remove outliers

In [4]:
# Drop outliers
min_price = 10
max_price = 350
idx = df['price'].between(min_price, max_price)
df = df[idx].copy()
# Convert last_review to datetime
df['last_review'] = pd.to_datetime(df['last_review'])

In [5]:
df

name    host_id  \
id                                                                       
9138664                 Private Lg Room 15 min to Manhattan   47594947   
31444015  TIME SQUARE CHARMING ONE BED IN HELL'S KITCHEN...    8523790   
8741020   Voted #1 Location Quintessential 1BR W Village...   45854238   
34602077  Spacious 1 bedroom apartment 15min from Manhattan  261055465   
23203149   Big beautiful bedroom in huge Bushwick apartment     143460   
...                                                     ...        ...   
7094539                   Tranquil haven in bubbly Brooklyn    2052211   
4424261                     Large 1 BR with backyard on UWS    3447311   
4545882                 Amazing studio/Loft with a backyard   23569951   
26518547          U2 comfortable double bed sleeps 2 guests     295128   
33631782               Private Bedroom in Williamsburg Apt!    8569221   

             host_name neighbourhood_group    neighbourhood  latitude  \
id                                                                      
9138664           Iris              Queens        Sunnyside  40.74271   
31444015        Johlex           Manhattan   Hell's Kitchen  40.76682   
8741020           John           Manhattan     West Village  40.73631   
34602077         Regan              Queens          Astoria  40.76424   
23203149         Megan            Brooklyn         Bushwick  40.69839   
...                ...                 ...              ...       ...   
7094539        Adriana            Brooklyn  Windsor Terrace  40.65360   
4424261          Sarah           Manhattan  Upper West Side  40.80188   
4545882          Kaveh           Manhattan  Upper East Side  40.78110   
26518547  Carol Gloria               Bronx     Clason Point  40.81225   
33631782          Andi            Brooklyn     Williamsburg  40.71829   

          longitude        room_type  price  minimum_nights  \
id                                                            
9138664   -73.92493     Private room     74               2   
31444015  -73.98878  Entire home/apt    170               3   
8741020   -74.00611  Entire home/apt    245               3   
34602077  -73.92351  Entire home/apt    125               3   
23203149  -73.92044     Private room     65               2   
...             ...              ...    ...             ...   
7094539   -73.97546  Entire home/apt    143              14   
4424261   -73.96808  Entire home/apt    200               2   
4545882   -73.94567  Entire home/apt    220               3   
26518547  -73.85502     Private room     80               1   
33631782  -73.95819     Private room    109               3   

          number_of_reviews last_review  reviews_per_month  \
id                                                           
9138664                   6  2019-05-26               0.13   
31444015                  0         NaT                NaN   
8741020                  51  2018-09-19               1.12   
34602077                  1  2019-05-24               0.65   
23203149                  8  2019-06-23               0.52   
...                     ...         ...                ...   
7094539                   2  2016-08-27               0.04   
4424261                  22  2019-05-21               0.50   
4545882                  28  2019-05-23               0.50   
26518547                  4  2019-07-01               1.48   
33631782                  3  2019-04-28               1.07   

          calculated_host_listings_count  availability_365  
id                                                          
9138664                                1                 5  
31444015                               1               188  
8741020                                1                 0  
34602077                               1                13  
23203149                               2                 8  
...                                  ...               ...  
7094539                    

## Check with df.info() that all obvious problems have been solved

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 19001 entries, 9138664 to 33631782
Data columns (total 15 columns):
 #   Column                          Non-Null Count  Dtype         
---  ------                          --------------  -----         
 0   name                            18994 non-null  object        
 1   host_id                         19001 non-null  int64         
 2   host_name                       18993 non-null  object        
 3   neighbourhood_group             19001 non-null  object        
 4   neighbourhood                   19001 non-null  object        
 5   latitude                        19001 non-null  float64       
 6   longitude                       19001 non-null  float64       
 7   room_type                       19001 non-null  object        
 8   price                           19001 non-null  int64         
 9   minimum_nights                  19001 non-null  int64         
 10  number_of_reviews               19001 non-null  int64        

From the results, we have some comments:
* The number of data rows decreased from 20000 to 19001.
* The last_review column has been converted to datetime format. 

## Terminate the run

In [5]:
run.finish()